### ISSUE TITLE
warp_perspective ignores align_corners=False in homography normalization (identity warp fails)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#3904]
Found while validating PR #3650. `warp_perspective` with an IDENTITY homography does not reproduce the input when `align_corners=False`, while `warp_affine` does:

```python
import torch
from kornia.geometry.transform import get_perspective_transform
from kornia.geometry.transform.imgwarp import warp_affine, warp_perspective

img = torch.arange(16.0).view(1, 1, 4, 4)
pts = torch.tensor([[[0.0, 0.0], [3.0, 0.0], [3.0, 3.0], [0.0, 3.0]]])
M = get_perspective_transform(pts, pts)  # identity

warp_affine(img, M[:, :2, :], (4, 4), align_corners=False)       # == img  (correct)
warp_perspective(img, M, (4, 4), align_corners=False)            # != img  (wrong)
# align_corners=True: both correct
```

Root cause: `normalize_homography` maps `[0, size-1]` (pixel centers at corners) to `[-1, 1]` unconditionally — the align_corners=True convention. Under `align_corners=False`, `grid_sample` treats ±1 as pixel EDGES, so the normalization needs the `[-0.5, size-0.5] → [-1, 1]` mapping; without it every `warp_perspective(..., align_corners=False)` samples with a subpixel scale/shift error (~half-pixel at the borders, grows with output size).

Impact: any `warp_perspective`/`homography_warp` call with `align_corners=False`; also blocks the otherwise-correct fix in #3650 (crop_and_resize for non-parallel quads), which switches crops from warp_affine to warp_perspective and inherits this error for the `align_corners=False` path — that is exactly the failure `TestCropAndResize::test_align_corners_false` catches (1.55 abs diff).

Candidate fix: make `normal_transform_pixel` (and its callers in `normalize_homography`/`warp_perspective`) respect `align_corners`, mirroring what warp_affine's path does. Needs a careful sweep since `normalize_homography` is public API with documented semantics (pixel → [-1,1]).

After fixing, add the pair `warp_affine`/`warp_perspective` identity-consistency check to the conventions page's align_corners section.

**Comment by VanniLeonardo** [comment]
Hi @ducha-aiki I would like to work on this. Can I be assigned?

Reproduced on current `main` (torch 2.9.1+cu128, Linux, RTX 3090):

```
warp_affine       max|diff| = 1.4e-05
warp_perspective  max|diff| = 11.25
```

Moreover I wanted to ask

**1.** `warp_image_tps` in #3928 gives the *same* `11.25` on the same 4×4 input, which fits both being the one `[0, size-1]` vs `[-0.5, size-0.5]` normalization mismatch — `normal_transform_pixel` here, `create_meshgrid` there. Would you prefer one PR covering both, or keep them separate?

**2.** For the fix itself: thread `align_corners` through `normal_transform_pixel` and `normalize_homography`, or add an internal helper and leave the public signatures alone? Since `normalize_homography` is public with documented semantics, I'd rather follow your call than guess.

**Comment by ducha-aiki** [comment]
Yes, please.  Let's cover them in the same PR.
I think the threading them is a correct way.

**Comment by VanniLeonardo** [comment]
Okay perfect, whenever you can assign me I'll start implementing
